presidio-analyzer/presidio_analyzer/pattern_recognizer.py

```diff
@@ -182,7 +182,11 @@ def __analyze_patterns(
         results = []
         for pattern in self.patterns:
             match_start_time = datetime.datetime.now()
-            matches = re.finditer(pattern.regex, text, flags=flags)
+            matches = re.finditer(
+                pattern.regex,
+                text,
+                flags=flags ^ re.I if pattern.case_sensitive else flags,
```

Comment:
Will this work if the user didn't pass re.I in the first place?

Reply:
Right! forgot my CS 101